src/ai/backend/manager/api/session.py

```diff
@@ -2178,6 +2180,111 @@ async def shutdown_service(request: web.Request, params: Any) -> web.Response:
     return web.Response(status=204)
 
 
+async def find_dependency_sessions(
+    root_session_name_or_id: Union[uuid.UUID, str],
+    db_connection: SAConnection,
+    access_key: AccessKey,
+):
+    dependency_session_cache: Dict[str, List] = {}
+
+    async def _find_dependency_sessions(session_name_or_id: Union[uuid.UUID, str]):
```

Comment:
1. I wanted to exclude `db_connection`, `access_key` from the arguments of the recursive function explicitly because they should be the same object as the initial argument in all recursive functions.

2. `dependency_session_cache` could be assigned when the `find_dependency_sessions` is called and could be accessed in all recursive calls without increasing argument counts. and it also removes passing an empty dictionary from the callsite.